pact/matchers.py

```diff
@@ -0,0 +1,196 @@
+        """
+        valid_types = (
+            type(None), list, dict, int, float, six.string_types, Matcher)
+
+        assert isinstance(matcher, valid_types), (
+            "matcher must be one of '{}', got '{}'".format(
+                valid_types, type(matcher)))
+
+        self.matcher = matcher
+
+    def generate(self):
+        """
+        Return the value that should be used in the request/response.
+
+        :return: A dict containing the information about what the contents of
+            the response should be.
+        :rtype: dict
+        """
+        return {
+            'json_class': 'Pact::SomethingLike',
+            'contents': self.matcher}
+
+
+class Term(Matcher):
+    """
+    Expect the response to match a specified regular expression.
+
+    Example:
+
+    >>> from pact import Consumer, Provider
+    >>> pact = Consumer('consumer').has_pact_with(Provider('provider'))
+    >>> (pact.given('the current user is logged in as `tester`')
+    ...  .upon_receiving('a request for the user profile')
+    ...  .with_request('get', '/profile')
+    ...  .will_respond_with(200, body={
+    ...    'name': 'tester',
+    ...    'theme': Term('light|dark|legacy', 'dark')
+    ...  }))
+
+    Would expect the response body to be a JSON object, containing the key
+    `name`, which will contain the value `tester`, and `theme` which must be
+    one of the values: light, dark, or legacy. When the consumer runs this
+    contract, the value `dark` will be returned by the mock service.
+    """
+
+    def __init__(self, matcher, generate):
+        """
+        Create a new Term.
+
+        :param matcher: A regular expression to find.
+        :type matcher: basestring
+        :param generate: A value to be returned by the mock service when
+            generating the response to the consumer.
+        :type generate: basestring
+        """
+        self.matcher = matcher
+        self._generate = generate
+
+    def generate(self):
+        """
+        Return the value that should be used in the request/response.
+
+        :return: A dict containing the information about what the contents of
+            the response should be, and what should match for the requests.
+        :rtype: dict
+        """
+        return {
+            'json_class': 'Pact::Term',
+            'data': {
+                'generate': self._generate,
+                'matcher': {
+                    'json_class': 'Regexp',
+                    'o': 0,
+                    's': self.matcher}}}
+
+
+def from_term(term):
+    """
+    Parse the provided term into the JSON for the mock service.
+
+    :param term: The term to be parsed.
+    :type term: None, list, dict, int, float, str, unicode, Matcher
+    :return: The JSON representation for this term.
+    :rtype: dict, list, str
+    """
+    if term is None:
+        return term
+    elif isinstance(term, (six.string_types, int, float)):
+        return term
+    if isinstance(term, dict):
```

Comment:
Nice catch. This one made it through all our internal reviews. With the return statements its functionally the same, but for consistency I think its best to change it to `elif`.